Goals
- Determine how others have benchmarked metagenomic analysis tools


**Metapallete** 

Data:
- Human Micrbiome Project (HMP) mock community even (NCBI  Accession number SRR172902) with 31 spiked in strains of *Lysinibacillus sphaericus* and two novel strains generated with Grinder 25. 
- HMP mock community + 50,000 simulated reads from 
- Metagenomic soil samples (i.e. Iowa pairie metagenomic samples (MG Rast #6377) 
- Simulated data from Lindgreen et al 2016

Benchmarking:


**Lindgreen comparative paper (14 tools)**

Data: 
- Simulated data set; 2 x 100 bp reads from real and simulated genomes. 
	- Rsesulting in 6 metagenomes containing between 27 and 37 million read pairs. 

Benchmarking: 
- Reads mapped 
- Shuffled reads mapped (Shuffled reads generated using HMMER shuffle package) 
- How many phyla were predicted that were not in the data?
- Comparison of predicted relative abundance of phyla to actual abundance
- Correlation of w/ known community composition using pearson correlation 
- Multivariate analysis - Observed agreement between 1) predictions generated by each tool and 2) the prediction from each tool and real distributions.   
- Sensitivity, specificity, positive predictive value, and Mathews correction. 
- Compared predicitons generated by 14 tools: CLARK16, Genometa17, GOTTCHA18, Kraken19, LMAT20, MEGAN21,22, MG-RAST23, the One Codex webserver, taxator-tk, MetaPhlAn25, MetaPhyler 26, mOTU27, QIIME) 

**Kraken** 

Data: 
- 3 simulated metagenomes 
	- 2 metagenomes generated by combining real reads from MiSeq and HiSeq
	- 1 metagenome consisting of simulated bacterial and archael reads with a error rate 5x's what would be expected (SimBA-5) 

Benchmarking: 
- Classfication accuracy 
	- Evaluated Krakens ability to assign sequences to the correct genus using 10,000 classified reads from a simulated data set (Sensitivity). 
	- Determine the number of correct classifications out of classifications attempted (precision).


The process

- [ ] one thing I'd really appreciate is a summary of how they've done validation in other papers, and what we need to prepare in order to do the same.

- 

References 

- Metapallete - David   Koslicki   , D. F. (2016). MetaPalette: aÂ k-mer Painting Approach for Metagenomic Taxonomic Profiling and Quantification of Novel Strain Variation, mSystems.

- Lindgreen, S., K. L. Adair and P. P. Gardner (2016). "An evaluation of the accuracy and speed of metagenome analysis tools." Sci Rep 6: 19233.

- Wood, D. E. and S. L. Salzberg (2014). "Kraken: ultrafast metagenomic sequence classification using exact alignments." Genome Biol 15(3): R46.


